src/test/java/seedu/apollo/command/task/SpecificHelpCommandTest.java

```diff
@@ -0,0 +1,182 @@
+package seedu.apollo.command.task;
+
+import org.junit.jupiter.api.Test;
+import seedu.apollo.calendar.Calendar;
+import seedu.apollo.command.Command;
+import seedu.apollo.command.utils.specifichelpcommand.DateHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.DeadlineHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.DeleteHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.EventHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.FindHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.ListHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.MarkHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.TodoHelpCommand;
+import seedu.apollo.command.utils.specifichelpcommand.UnmarkHelpCommand;
+import seedu.apollo.module.ModuleList;
+import seedu.apollo.storage.Storage;
+import seedu.apollo.task.TaskList;
+import seedu.apollo.ui.Parser;
+import seedu.apollo.ui.Ui;
+
+import java.io.FileNotFoundException;
+import java.rmi.UnexpectedException;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class SpecificHelpCommandTest {
+    TaskList taskList = new TaskList();
+    Ui ui = new Ui();
+    Storage storage = new Storage("test.txt", "testModuleData.txt");
+    ModuleList moduleList = new ModuleList();
+    ModuleList allModules = storage.loadModuleData();
+    Calendar calendar = new Calendar();
+
+    public SpecificHelpCommandTest() throws FileNotFoundException {
+
+    }
+
+    @Test
+    void parseHelpCommand_invalidCommand_expectException() throws IllegalArgumentException {
+        String userCommand = "help draw";
+        assertThrows(IllegalArgumentException.class, () -> Parser.chooseHelpCommand(userCommand));
+    }
+
+
+    @Test
+    void parseHelpCommand_listHelpCommand_expectNoException() throws UnexpectedException {
+
+
+        String userCommand = "help list";
+        Ui ui = new Ui();
+        int size = 1;
+        Command newCommand = Parser.getCommand(userCommand, ui, size, null);
+        assertEquals(ListHelpCommand.class, newCommand.getClass());
+        ListHelpCommand newListHelpCommand = new ListHelpCommand();
+        assertDoesNotThrow(() ->
+                newListHelpCommand.execute(taskList, ui, storage, moduleList, allModules, calendar));
```

Comment:
why not just execute the newCommand instead of creating a newListHelpCommand since newCommand should already be a ListHelpCommand?

Reply:
same for all other tests

Reply:
also Ui ui can just be created within the class for SpecificHelpCommandTest so you don't have to keep declaring it in every test